crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -418,77 +419,99 @@ fn get_dependency_hashes_from_ops(ops: impl Iterator<Item = DhtOpHashed>) -> Vec
-                            RecordEntry::Present(entry @ Entry::CounterSign(session_data, _)) => {
-                                // Discard errors here because we'll check later whether the input is valid. If it's not then it
-                                // won't matter that we've skipped fetching deps for it
-                                if let Ok(entry_rate_weight) = action_to_entry_rate_weight(action) {
+                DhtOp::ChainOp(op) => {
+                    let op_dependency_hashes = match &**op {
+                        ChainOp::StoreRecord(_, action, entry) => {
+                            let mut actions = match entry {
+                                RecordEntry::Present(
+                                    entry @ Entry::CounterSign(session_data, _),
+                                ) => {
+                                    // Discard errors here because we'll check later whether the input is valid. If it's not then it
+                                    // won't matter that we've skipped fetching deps for it
+                                    if let Ok(entry_rate_weight) =
+                                        action_to_entry_rate_weight(action)
+                                    {
+                                        make_action_set_for_session_data(
+                                            entry_rate_weight,
+                                            entry,
+                                            session_data,
+                                        )
+                                        .unwrap_or_else(|_| vec![])
+                                        .into_iter()
+                                        .map(|action| action.into_hash())
+                                        .collect::<Vec<_>>()
+                                    } else {
+                                        vec![]
+                                    }
+                                }
+                                _ => vec![],
+                            };
+
+                            if let Action::Update(update) = action {
+                                actions.push(update.original_action_address.clone());
+                            }
+                            Some(actions)
+                        }
+                        ChainOp::StoreEntry(_, action, entry) => {
+                            let mut actions = match entry {
+                                Entry::CounterSign(session_data, _) => {
+                                    // Discard errors here because we'll check later whether the input is valid. If it's not then it
+                                    // won't matter that we've skipped fetching deps for it
                                     make_action_set_for_session_data(
-                                        entry_rate_weight,
+                                        new_entry_action_to_entry_rate_weight(action),
```

Comment:
What's this change about? How is it related?

Reply:
I haven't changed this. The diff is confusing, in this section only the indentation should have changed.